writer.go

```diff
@@ -1232,20 +1235,29 @@ func (c *writerColumn) flushFilterPages() error {
 		index:              int16(c.bufferIndex),
 	}
 
-	rbuf, pool := getBufioReader(nil, 1024)
+	var pageReader io.Reader = c.pageBuffer
+	if _, err = c.pageBuffer.Seek(0, io.SeekStart); err != nil {
```

Comment:
Do we want to check if the returned `offset` == 0 as in the previous implementation?

Reply:
I can add that back, though I was confused by those checks - if Seek(0, io.SeekStart) didn't set the offset to zero, shouldn't an error have occurred?

Reply:
Not sure what the intent was here, maybe @achille-roussel has more context on why the explicit check was in place.

Reply:
I think the check was just to ensure that custom implementation of the interface behave as expected, it was a safeguard but shouldn't affect the logic in any other ways.